account/watcher/watcher.go

```diff
@@ -0,0 +1,103 @@
+package watcher
+
+import (
+	"sync"
+
+	"github.com/btcsuite/btcd/btcec"
+)
+
+// expiryWatcher implements the ExpiryWatcher interface
+type expiryWatcher struct {
+	handlers EventHandler
+
+	// bestHeight is the height we believe the current chain is at.
+	bestHeight uint32
+
+	// expirations keeps track of the current accounts we're
+	// watching expirations for.
+	expirations map[[33]byte]uint32
+
+	// expirationsPerHeight keeps track of all registered accounts
+	// that expire at a certain height.
+	expirationsPerHeight map[uint32][]*btcec.PublicKey
+
+	expirationsMtx sync.Mutex
+}
+
+// NewExpiryWatcher instantiates a new ExpiryWatcher.
+func NewExpiryWatcher(handlers EventHandler) *expiryWatcher { // nolint:golint
+	return &expiryWatcher{
+		handlers:             handlers,
+		expirations:          make(map[[33]byte]uint32),
+		expirationsPerHeight: make(map[uint32][]*btcec.PublicKey),
+	}
+}
+
+// NewBlock updates the current bestHeight.
+func (w *expiryWatcher) NewBlock(bestHeight uint32) {
+	w.bestHeight = bestHeight
+}
+
+// OverdueExpirations handles the expirations for the given block.
+func (w *expiryWatcher) OverdueExpirations(blockHeight uint32) {
+	w.expirationsMtx.Lock()
+	defer w.expirationsMtx.Unlock()
+
+	for _, traderKey := range w.expirationsPerHeight[blockHeight] {
+		var accountKey [33]byte
+		copy(accountKey[:], traderKey.SerializeCompressed())
+
+		// If the account doesn't exist within the
+		// expiration set, then the request was
+		// canceled and there's nothing for us to do.
+		// Similarly, if the request was updated to
+		// track a new height, then we can skip it.
+		curExpiry, ok := w.expirations[accountKey]
+		if !ok || blockHeight != curExpiry {
+			continue
+		}
+
+		err := w.handlers.HandleAccountExpiry(
```

Comment:
Do we need to lock for `HandleAccountExpiry`? An alternative could be to collect expiring accounts in the lock then expire them outside the lock.